### PR TITLE
Added basic methods for DownloadStation.API

### PR DESCRIPTION
**Pull request by ikocs**
We propose the implementation of the following methods:

- Info:
  - Info;
  - SetServerConfig;
  - GetConfig;
- Task
  - Create (due to bugs in Synology, it is only possible to add from the link);
  - Delete;
  - Pause;
  - Resume.

### CODE REVIEW

**Comment by plneto** [comment]
Thanks for submitting this PR. I'll try to review it soon.

I noticed that you added some PDFs to the repository which I think are the same ones that are already in the `docs` folder. Can you please check and remove them if they are duplicates?

**Comment by ikocs** [comment]
> I noticed that you added some PDFs to the repository which I think are the same ones that are already in the `docs` folder. Can you please check and remove them if they are duplicates?

fix it

**Comment by chesh111re** [comment]
`ISynologyHttpClient` has `GetAsync` method with `queryParams`. It had type `Dictionary<string, string>`
I made `Dictionary<string, string?>` because some parameters are optional

File: src/Synology.Api.Client.Integration.Tests/DownloadStationApiTests.cs [hunk 1]
```diff
@@ -28,5 +33,228 @@ public async void DownloadStationApi_DownloadStation_ListTasks()
...
+
+            var pauseResponse = _fixture
+                .Client
+                .DownloadStationApi()
+                .TaskEndpoint()
+                .PauseAsync(listTask.Result.Tasks.Last().Id);
+
+            pauseResponse.Should().NotBeNull();
+            
+            var delRequest = new DownloadStationTaskDeleteRequest
```
Comment (plneto) [inline src/Synology.Api.Client.Integration.Tests/DownloadStationApiTests.cs:172]: Did you forget to execute the delete request?

File: src/Synology.Api.Client/Apis/DownloadStation/Info/IDownloadStationInfoEndpoint.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+using Synology.Api.Client.Apis.DownloadStation.Info.Models;
+using Synology.Api.Client.Shared.Models;
+
+namespace Synology.Api.Client.Apis.DownloadStation.Info
+{
+    public interface IDownloadStationInfoEndpoint
+    {
+        Task<DownloadStationInfoResponse> InfoAsync();
+        
+        Task<DownloadStationServerConfig> GetConfigAsync();
```
Comment (plneto) [inline src/Synology.Api.Client/Apis/DownloadStation/Info/IDownloadStationInfoEndpoint.cs:11]: Can you please add the XML documentation to these 2 methods as well?

File: src/Synology.Api.Client/Apis/DownloadStation/Info/Models/DownloadStationServerConfig.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
+﻿using System.Text.Json.Serialization;
+
+namespace Synology.Api.Client.Apis.DownloadStation.Info.Models
+{
+    public class DownloadStationServerConfig
+    {
+        public DownloadStationServerConfig(
```
Comment (plneto) [inline src/Synology.Api.Client/Apis/DownloadStation/Info/Models/DownloadStationServerConfig.cs:7]: What's the expected behaviour when you set only one config and all the other values are null? Are the ones with null values ignored by the DSM?

File: src/Synology.Api.Client/SynologyHttpClient.cs [hunk 4]
```diff
@@ -93,19 +93,19 @@ private async Task<T> HandleSynologyResponse<T>(HttpResponseMessage httpResponse
                 case HttpStatusCode.OK:
                     {
                         var response = await httpResponse.Content.ReadFromJsonAsync<ApiResponse<T>>();
-                        if (!response.Success)
+                        if (!response?.Success ?? true)
```
Comment (plneto) [inline src/Synology.Api.Client/SynologyHttpClient.cs:96]: Maybe we can change this line to make it easier to understand? Something like `if (!(response?.Success ?? false))`

**Review by plneto — CHANGES_REQUESTED** [review]
Looks good to me. Please fix the minor issues from the comments I left.

**Comment by ikocs** [comment]
Good afternoon In the near future there will be no time at all to look at and correct the comments. But I hope to return to the project soon.

**Comment by plneto** [comment]
> Good afternoon In the near future there will be no time at all to look at and correct the comments. But I hope to return to the project soon.

No worries, I'll merge the PR. They are minor issues so you can fix them later.

**Pull request merged**